library/func_misc.py

```diff
@@ -153,3 +153,55 @@ def Call(self, pos_args, named_args):
 
         raise error.TypeErr(x, 'len() expected Str, List, or Dict',
                             loc.Missing)
+
+
+class Join(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+
+        r = typed_args.Reader(pos_args, named_args)
+        li = r.PosList()
+
+        delim = r.NamedStr('delim', '')
+        if len(pos_args): # reader has a reference
+            delim = r.PosStr()
+
+        r.Done()
+
+        strs = []  # type: List[str]
+        for i, el in enumerate(li):
+            msg = 'join() items must be Str but item %d is %s' % (
+                i, value_str(el.tag()))
+            strs.append(val_ops.ToStr(el, msg, loc.Missing))
+
+        return value.Str(delim.join(strs))
+
+
+class Maybe(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+
+        r = typed_args.Reader(pos_args, named_args)
+        val = r.PosValue()
+        r.Done()
+
+        if val == value.Undef or val == value.Null:
```

Comment:
Undef should only be used for signaling a variable is undefined, so it won't be passed to a function

You could add `assert val != value.Undef` to be sure